sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/Plan.java

```diff
@@ -44,4 +47,8 @@ default boolean isRecoveryPlan() {
     default boolean isDecommissionPlan() {
         return getName().equals(DECOMMISSION_PLAN_NAME);
     }
+
+    Optional<String> getNamespace();
+
+    Collection<Protos.TaskInfo> getExistingTasks();
```

Comment:
What calls this? I didn't see anything using it in the PR but could've missed something.

Could the user of this information just query the StateStore directly? The tasks passed into the Plan at construction could fall out of date.

Reply:
Much better idea.